src/utils/extractTestResults.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+import { DynamoDBRecord } from 'aws-lambda';
+import { DynamoDB } from 'aws-sdk';
+import { DateTime } from 'luxon';
+import { TestResult } from './testResult';
+import { MCRequest } from './MCRequest';
+import logger from '../observability/logger';
+
+/**
+ * This is used to extract the relevant fields from the test record that is
+ * required to be sent to MC in order to  clear prohibitions
+ * @param record
+ */
+export const extractMCTestResults = (record: DynamoDBRecord): MCRequest[] => {
+  try {
```

Comment:
Is there anything in this `try` that could throw an error?

Reply:
I put the try catch here as a fail safe for when it's extracting and constructing the new request object. In this case for whatever reason, the record coming in didn't have a vin field, it would throw an error when trying to do the following data.vin or it could throw an error when trying to format the date

Reply:
In the case of accessing an `undefined` property like `data.vin`, it wouldn't throw an error it would just assign `undefined` to the object attribute your map returns. W.r.t. formatting a date, it might be worth writing a test passing an invalid date and seeing what it does (if you haven't already), from a quick skim of the luxon docs I couldn't see how it handles invalid inputs

Reply:
But with the data coming from VTA/VTM -> Dynamo where the data is already validated to be a date, is it safe to assume that your data will always be valid?

Reply:
That's fair enough, i'll remove the try catch thanks

Reply:
I do believe there should be some validation somewhere to ensure that the payload you are sending to search MC matches what they're expecting, i.e. no `undefined` values which wouldn't have been caught by your `try catch`